pkg/users/group-create-dialog.js

```diff
@@ -0,0 +1,169 @@
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React from 'react';
+
+import { Form, FormGroup, TextInput } from '@patternfly/react-core';
+import { show_modal_dialog, apply_modal_dialog } from "cockpit-components-dialog.jsx";
+
+import { has_errors, is_valid_char_name } from "./dialog-utils.js";
+
+const _ = cockpit.gettext;
+
+function GroupCreateBody({ state, errors, change }) {
+    const {
+        name, id,
+    } = state;
+
+    return (
+        <Form isHorizontal onSubmit={apply_modal_dialog}>
+            <FormGroup label={_("Name")}
+                       helperTextInvalid={errors && errors.name}
+                       validated={(errors && errors.name) ? "error" : "default"}
+                       fieldId="groups-create-name">
+                <TextInput id="groups-create-name"
+                           validated={(errors && errors.name) ? "error" : "default"}
+                           value={name} onChange={value => change("name", value)} />
+            </FormGroup>
+
+            <FormGroup label={_("ID")}
+                       hasNoPaddingTop
+                       helperTextInvalid={errors && errors.id}
+                       validated={(errors && errors.id) ? "error" : "default"}
+                       isStack
+                       fieldId="groups-create-id">
+                <TextInput id="groups-create-id"
+                           validated={(errors && errors.id) ? "error" : "default"}
+                           value={id} onChange={value => change("id", value)} />
+            </FormGroup>
+        </Form>
+    );
+}
+
+function validate_name(name, groups) {
+    if (!name)
+        return _("No group name specified");
+
+    for (let i = 0; i < name.length; i++) {
+        if (!is_valid_char_name(name[i]))
+            return _("The group name can only consist of letters from a-z, digits, dots, dashes and underscores.");
+    }
+
+    for (let k = 0; k < groups.length; k++) {
+        if (groups[k].name == name)
+            return _("A group with this name already exists");
+    }
+
+    return null;
+}
+
+function validate_group(id, groups) {
+    if (!id)
+        return _("No ID specified");
+
+    const id_num = parseInt(id);
+    if (id_num.toString() !== id || id_num < 0)
+        return _("The group ID must be positive integer.");
+
+    for (let k = 0; k < groups.length; k++) {
+        if (groups[k].gid == id_num)
+            return _("A group with this ID already exists");
```

Comment:
Please pick an error message style -- either with a period or without. Here, in validate_name(), and presumably other places there are both variants.

Reply:
Without period is the PF suggestion. Removed periods.